### PR TITLE
refactor: refactor the switch statement to MAP

### PR DESCRIPTION
**Pull request by Paulo-Lopes-Estevao**
This PR was created in order to add the feature suggested here in this ISSUE [#3](https://github.com/antonio-nicolau/ClientAO/issues/3)

### CODE REVIEW

File: lib/src/shared/repositories/api_request.repository.dart [hunk 1]
```diff
@@ -29,25 +28,21 @@ class ApiRequest implements IApiRequestRepository {
...
-        break;
+    
+    final methodHandlers = <HttpVerb, Future<Response>>{
+      HttpVerb.get: _httpClient.get(uri!, headers: headers),
+      HttpVerb.post: _httpClient.post(uri!, headers: headers, body: body),
+      HttpVerb.put: _httpClient.put(uri!, headers: headers, body: body),
+      HttpVerb.patch: _httpClient.patch(uri!, headers: headers, body: body),
+      HttpVerb.delete: _httpClient.delete(uri!, headers: headers, body: body),
+  };
+
```
Comment (antonio-nicolau) [inline lib/src/shared/repositories/api_request.repository.dart:39]: we can also export that into a separated method e.g `doRequest` giving corrects params like `httpClient, method, uri, headers and body`

See my comment here [issue 3](https://github.com/antonio-nicolau/ClientAO/issues/3#issuecomment-1523075830)

File: lib/src/shared/repositories/api_request.repository.dart [hunk 2]
```diff
@@ -29,25 +28,21 @@ class ApiRequest implements IApiRequestRepository {
...
+    if (!methodHandlers.containsKey(request.method)) {
+      throw UnsupportedError('Unsupported HTTP method: ${request.method}');
     }
 
+    final response = await methodHandlers[request.method]!;
```
Comment (antonio-nicolau) [inline lib/src/shared/repositories/api_request.repository.dart:44]: we will end up having 

```dart
final response = await doRequest(method:request.method,
      httpClient: _httpClient,
      uri: uri,
      headers: headers,
      body: body,
      );
```

**Review by antonio-nicolau — CHANGES_REQUESTED** [review]
Nice one :) I left my thoughts there 😉

**Review by IldySilva — COMMENTED** [review]
more readability/maintainability for code, **nice approach**

**Comment by Paulo-Lopes-Estevao** [comment]
> Nice one :) I left my thoughts there 😉

I added the idea of reuse using the "doRequest" method.
It is a great idea 👍🏼.

**Pull request merged**